File: tools/samples/dotnet/Services/Pricing.cs
using Sample.Domain;

namespace Sample.Services;

public static class Pricing
{
    /// <summary>Puts a basket total into a size band.</summary>
    public static string Classify(decimal total)
    {
        if (total > 100m)
        {
            return "large";
        }

        if (total > 50m)
        {
            return "medium";
        }

        if (total > 10m)
        {
            return "small";
        }

        if (total > 0m)
        {
            return "tiny";
        }

        return "empty";
    }

    public static string Band(Basket basket, decimal discount)
    {
        var total = basket.Total(discount);
        if (total < 0m)
            return "refund";

        return Classify(total);
    }

    private static decimal WithVat(decimal value)
    {
        return value * 1.2m;
    }
}

File: tools/samples/java/src/main/java/com/example/services/Pricing.java
package com.example.services;

import com.example.domain.Basket;

public final class Pricing {
    private Pricing() {
    }

    /** Puts a basket total into a size band. */
    public static String classify(int total) {
        if (total > 100) {
            return "large";
        }
        if (total > 50) {
            return "medium";
        }
        if (total > 10) {
            return "small";
        }
        if (total > 0) {
            return "tiny";
        }
        return "empty";
    }

    public static String band(Basket basket, int discount) {
        int total = basket.total(discount);
        if (total < 0)
            return "refund";
        return classify(total);
    }

    private static int withVat(int value) {
        return value * 12 / 10;
    }
}
